README.rst

```diff
@@ -32,58 +35,83 @@ On some networks discovery doesn't work reliably, in that case if you can find t
-Setting `wifi=True` will clear wifi information ("Change Wi-Fi" in the Wemo app), which does not clear the rules, name, etc.
-Setting both to true is equivalent to a "Factory Restore" from the app.
-It should also be noted that devices contain a hardware reset procedure as well, so using the software is for convenience or if physical access is not available.
+Reset
+~~~~~
+Reset can be performed with the ``reset`` method, which has 2 boolean input arguments, ``data`` and ``wifi``.
+WeMo devices contain a hardware reset procedure as well, so use of ``pywemo`` is for convenience or if physical access is not available.
+This ``reset`` method may not work on all devices.
+
+=======================================  =================  =======================
+Method in ``pywemo``                     Clears             Name in WeMo App
+=======================================  =================  =======================
+``device.reset(data=True, wifi=False)``  name, icon, rules  Clear Personalized Info
+``device.reset(data=False, wifi=True)``  wifi information   Change Wi-Fi
+``device.reset(data=True, wifi=True)``   everything         Factory Restore
+=======================================  =================  =======================
+
+Setup
+~~~~~
 
-Device setup is through the `setup` method.
-The user must first connect to the devices locally broadcast access point, then discover the device there.
-Once done, pass the desired SSID and password (AES encryption only) to the `setup` method to connect it to your wifi network.
+Device setup is through the ``setup`` method, which has two required arguments: ``ssid`` and ``password``.
+The user must first connect to the devices locally broadcast access point, which typically starts with "WeMo.", and then discover the device there.
+Once done, pass the desired SSID and password (AES encryption only) to the ``setup`` method to connect it to your wifi network.
 
-Important Note for Device Setup - OpenSSL is Required!
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``device.setup(ssid='wifi_name', password='special_secret')``
 
-OpenSSL is used to encrypt the password by the pywemo library.
-It must be installed and available on the path via calling `openssl` with a terminal (or command prompt, if on Windows).
-This is not required if connecting the device to an open network, since that requires no password, although an open network certainly isn't recommended.
+A few important notes:
+
+- Not all devices are supported for setup.
+- WeMo requires a password of at least 8 characters long.
+- For a WeMo without internet access, see `this guide <https://github.com/pavoni/pywemo/wiki/WeMo-Cloud#disconnecting-from-the-cloud>`_ to stop any blinking lights.
+- If connecting to an open network, the password argument is ignored and you can provide anything, e.g. ``password=None``.
+- If connecting to an AES-encrypted network, OpenSSL is used to encrypt the password by the ``pywemo`` library.
```

Comment:
Sure, I'll change AES to WPA2, you're right that most people will be more familiar with that terminology.   As for the password length, I have no idea.  You're probably right, I've never tried a shorter password for my network.  Should I just remove that bullet then since that will always be satisfied if they are using WPA2?

Reply:
Ya, length 8 is a minimum for WPA2.  I'll just remove that bullet

Reply:
>Should I just remove that bullet then since that will always be satisfied if they are using WPA2?

Removing that part SGTM. I've been wondering if the `ShortPassword` exception is really an indicator of `IncorrectPassword`. I'd need to do more testing to verify that.